test/test-suite/test_foreign.py

```diff
@@ -254,6 +254,27 @@ def jpeg_valid(im):
             # format area at the end
             assert y.startswith("hello world")
 
+    @skip_if_no("jpegload")
+    def test_jpegsave(self):
+        im = pyvips.Image.new_from_file(JPEG_FILE)
+
+        # higher Q should mean a bigger buffer
+        q10 = im.jpegsave_buffer(Q=10)
+        q10_subsample_off = im.jpegsave_buffer(Q=10, subsample_mode=2)
+        q10_subsample_no = im.jpegsave_buffer(Q=10, no_subsample=1)
```

Comment:
I don't think we should have deprecated API being tested in the test suite, let's remove this.

Same for the case below. Just test `on`, `off`, `auto`.

Reply:
done